src/som/interpreter/nodes/nary/RedirectWriteNode.java

```diff
@@ -0,0 +1,49 @@
+package som.interpreter.nodes.nary;
+import com.oracle.truffle.api.frame.FrameSlot;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.compiler.Variable;
+import som.interpreter.nodes.ExpressionNode;
+import som.interpreter.nodes.PreevaluatedExpression;
+
+public class RedirectWriteNode extends ExprWithTagsNode
+    implements PreevaluatedExpression {
+
+  final VirtualFrame writeFrame;
+  final FrameSlot slot;
+  final ExpressionNode[] args;
+
+  public RedirectWriteNode(final VirtualFrame frame, final Variable var,
+      final ExpressionNode[] args, final SourceSection source) {
+    super(source);
+    this.writeFrame = frame;
+    this.args = args;
+
+    FrameSlot slotForVar = null;
+    for (FrameSlot slot : writeFrame.getFrameDescriptor().getSlots()) {
```

Comment:
The variable class contains behavior to read the value of variable from a frame, which I used to [evaluate and return ](https://github.com/richard-roberts/SOMns/blob/implement/objectLiterals/src/som/interpreter/nodes/nary/RedirectReadNode.java#L22) the value for the RedirectReadNode. There is no such behavior define to write a value to the frame (that I could find). The VirtualFrame implements a "setObject" method, which takes a  FrameSlot and a value. I need the lookup to find the FrameSlot that correlates to the variable I have.

Should I instead add behavior to the that handles writing in a similar way to [reading](https://github.com/richard-roberts/SOMns/blob/implement/objectLiterals/src/som/compiler/Variable.java#L142)?